### PR TITLE
Add Variables to Allow GitLab to use LetsEncrypt

### PR DESCRIPTION
**Pull request by williampiv**
- Adds basic functionality in configuration template to configure Gitlab to use Let'sEncrypt for SSL
- updates default variables (defaults/main.yml) to take into account these new variables

### CODE REVIEW

**Comment by williampiv** [comment]
- Appears Travis `debian9` is failing to update apt `TASK [Update apt cache.]`
- All other checks complete

File: defaults/main.yml [hunk 1]
```diff
@@ -58,3 +58,11 @@ gitlab_email_enabled: "false"
 gitlab_email_from: "gitlab@example.com"
 gitlab_email_display_name: "Gitlab"
 gitlab_email_reply_to: "gitlab@example.com"
+
+# LetsEncrypt configuration
+gitlab_letsencrypt: false
+gitlab_letsencrypt_contact_email: "gitlab@example.com"
```
Comment (geerlingguy) [inline defaults/main.yml:64]: See #148 — this should be `_emails` and allow a list, and then when it's printed you could use the `| to_json` filter so it's printed in a way that works in the ruby config file.

File: defaults/main.yml [hunk 2]
```diff
@@ -58,3 +58,11 @@ gitlab_email_enabled: "false"
 gitlab_email_from: "gitlab@example.com"
 gitlab_email_display_name: "Gitlab"
 gitlab_email_reply_to: "gitlab@example.com"
+
+# LetsEncrypt configuration
+gitlab_letsencrypt: false
```
Comment (geerlingguy) [inline defaults/main.yml:63]: Please wrap the `false` in strings (so `"false"`).

File: templates/gitlab.rb.j2 [hunk 3]
```diff
@@ -19,6 +19,15 @@ nginx['redirect_http_to_https'] = {{ gitlab_redirect_http_to_https }}
 nginx['ssl_certificate'] = "{{ gitlab_ssl_certificate }}"
 nginx['ssl_certificate_key'] = "{{ gitlab_ssl_certificate_key }}"
 
+{% if gitlab_letsencrypt == true %}
```
Comment (geerlingguy) [inline templates/gitlab.rb.j2:22]: Can drop the `== true`. Also, maybe always print the next line ('enable' line) and put this if condition on the line after?

File: templates/gitlab.rb.j2 [hunk 4]
```diff
@@ -19,6 +19,15 @@ nginx['redirect_http_to_https'] = {{ gitlab_redirect_http_to_https }}
...
+letsencrypt['enable'] = true
```
Comment (geerlingguy) [inline templates/gitlab.rb.j2:23]: After the earlier mentioned changes, this line can be:

```
letsencrypt['enable'] = "{{ gitlab_letsencrypt_enable }}"
```

File: defaults/main.yml (hunk 2 above)
Comment (geerlingguy) [reply defaults/main.yml:63]: Also call the variable `gitlab_letsencrypt_enable`

**Comment by aklinkert** [comment]
@geerlingguy Any chance to get this one merge somewhat soon? :)

**Comment by To1ne** [comment]
@geerlingguy Friendly reminder to have another look. As far as I can tell, all of your comments are addressed.

**Comment by Mizipzor** [comment]
Just came across this, I need to have letsencrypt disabled. Do you need any help completing this pull request?

**Comment by geerlingguy** [comment]
Sorry about the delay here, this totally fell off my radar!

**Review by geerlingguy — APPROVED** [review]

**Pull request merged**